invenio_rdm_records/services/generators.py

```diff
@@ -288,6 +288,14 @@ def needs(self, record=None, file_key=None, **kwargs):
         return []
 
 
+class IfCreate(ConditionalGenerator):
+    """Check if the request is created or modified."""
+
+    def _condition(self, record=None, request=None, **kwargs):
+        """Check if record is empty - meaning it is created for the first time ."""
+        return record is None and request is None
```

Comment:
I will add a comment, i think i have described it in another place
it is because in the services and permissions we pass usually `record=community`, `record=record`, etc, assuming that the parent classes and all the dependent classes understand record as an argument. However, for some reason we decided to make `request=request` in the requests service, and there are some classes in permissions expecting `record` as an argument and all hell breaks loose.
I have to find all the places we have `request` instead of `record` as a name of the argument and normalise it